upload/helpers.py

```diff
@@ -495,20 +495,6 @@ def insert_commit(commitid, branch, pr, repository, owner, parent_commit_id=None
             "state": "pending",
         },
     )
-
-    edited = False
-    if commit.state != "pending":
-        commit.state = "pending"
-        edited = True
-    if parent_commit_id and commit.parent_commit_id is None:
-        commit.parent_commit_id = parent_commit_id
-        edited = True
-    if branch and commit.branch != branch:
-        # A branch head may have been moved; this allows commits to be "moved"
-        commit.branch = branch
-        edited = True
```

Comment:
do you still want to keep these updates? these should be very infrequent I believe

Reply:
Regardless of frequency (though I agree), I think I should leave them for correctness. The new endpoint updates branch/parent as well.